src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -644,6 +665,8 @@ private static uint HandleEventShutdownComplete(State state)
                 state.ShutdownWriteCompletionSource.TrySetResult();
             }
 
+            state.ShutdownCompletionSource.TrySetResult();
```

Comment:
Is it possible for this event to be triggered more than once? Can we assert that `TrySetResult` succeeded?

Reply:
I've added a state that's changed under a lock, like ShutdownWriteCompletionSource has. After that I guess "Try" is not needed anymore (for ShutdownWriteCompletionSource too) -- we can just use SetResult